testcontainers/src/runners/async_runner.rs

```diff
@@ -20,6 +20,25 @@ use crate::{
 };
 
 const DEFAULT_STARTUP_TIMEOUT: Duration = Duration::from_secs(60);
+#[cfg(feature = "reusable-containers")]
+static TESTCONTAINERS_SESSION_ID: std::sync::OnceLock<ulid::Ulid> = std::sync::OnceLock::new();
+
+#[doc(hidden)]
+/// A unique identifier for the currently "active" `testcontainers` "session".
+///
+/// This identifier is used to ensure that the current "session" does not confuse
+/// containers it creates with those created by previous runs of a test suite.
+///
+/// For reference: without a unique-per-session identifier, containers created by
+/// previous test sessions that were marked `reuse`, (or where the test suite was
+/// run with the `TESTCONTAINERS_COMMAND` environment variable set to `keep`), that
+/// *haven't* been manually cleaned up could be incorrectly returned from methods
+/// like [`Client::get_running_container_id`](Client::get_running_container_id),
+/// as the container name, labels, and network would all still match.
+#[cfg(feature = "reusable-containers")]
+pub(crate) fn session_id() -> &'static ulid::Ulid {
+    TESTCONTAINERS_SESSION_ID.get_or_init(ulid::Ulid::new)
+}
```

Comment:
Not sure I get the `session_id` intent here - I mean it's definitely can be useful (for example to sort them if many)

But why do we want to filter by `session_id` label? 
The idea of reusable containers is to be able to use in next test run again

E.g: 
```sh
cargo test  # `session1`: creates containers
# do something
cargo test # `session2`: reuses containers of `session1`
```

Moreover, I might want to run parallel test suites with the same containers (I explicitly enabled re-usability)

Reply:
Ah - this *originally* came about after I ran into some trouble with the test suite failing and subsequently trying to reuse some containers *incorrectly* and then causing failures on *subsequent* runs.

I think a unique session identifier *is* a good idea / feature, memory is failing me as to why I didn't this up the same parallel test suite edge case you did though 🤔.

In either case, what do you think about keeping the ability, but making it's inclusion in filtration user-configurable? e.x.:

```rust
let reusable_container = GenericImage::new("alpine", "latest").with_reuse(true).start().await?;

let session_unique_container = GenericImage::new("alpine", "latest").with_reuse(true).current_session(true).start().await?;
```

That, or change the type of `ContainerRequest.reuse` from `bool` to something like `Option<ReuseDirective>`, e.x.:

```rust
pub enum ReuseDirective {
    Always,
    CurrentSession,
}
```

Reply:
I think `enum` option is a perfect fit here actually, because I see all of these options useful

Reply:
Yup, the more I thought about it after leaving the comment, the more I liked the idea.

Implemented it, pushed the changes. Please take a look at your convenience and let me know what you think 😁